crates/oxc_linter/src/rules/eslint/no_array_constructor.rs

```diff
@@ -62,10 +63,17 @@ impl Rule for NoArrayConstructor {
                 &new_expr.type_parameters,
                 false,
             ),
-            _ => return,
+            _ => {
+                return;
+            }
         };
 
-        if callee.is_specific_id("Array")
+        let Expression::Identifier(ident) = &callee else {
+            return;
+        };
+
+        if ctx.semantic().is_reference_to_global_variable(ident)
+            && callee.is_specific_id("Array")
```

Comment:
two small notes: 1) we should do the global check last (after all other checks), as it is more expensive, 2) we can use the `Expression.is_global_reference_name` method helper to make this a bit shorter:

```suggestion
        if callee.is_global_reference_name("Array", ctx.symbols())
```

Reply:
Thanks!

Reply:
Since callee cannot call `is_global_reference_name` directly, I made a slight modification. [ac35515](https://github.com/oxc-project/oxc/pull/7067/commits/ac35515d7c8a210919cb5720cc3cc3431a9b15a3)